eZ/Publish/Core/Search/Legacy/Content/WordIndexer/Gateway/DoctrineDatabase.php

```diff
@@ -0,0 +1,416 @@
+        SearchIndex $searchIndex
+    ) {
+        $this->dbHandler = $dbHandler;
+        $this->typeHandler = $typeHandler;
+        $this->transformationProcessor = $transformationProcessor;
+        $this->searchIndex = $searchIndex;
+        $this->dbMaxInt = $this->getDbMaxInt($dbHandler->getName());
+    }
+
+    /**
+     * Indexes an array of documents.
+     *
+     * Documents are given as an array of the array of documents. The array of documents
+     * holds documents for all translations of the particular entity.
+     *
+     * Notes:
+     * - On large amounts of data make sure to iterate with several calls to this function with a limited
+     *   set of documents, amount you have memory for depends on server, size of documents, & PHP version.
+     *
+     * @param \eZ\Publish\SPI\Search\Document[][] $documents
+     */
+    public function bulkIndexDocuments(array $documents)
+    {
+        foreach ($documents as $documentsChunk) {
+            foreach ($documentsChunk as $translationDocuments) {
+                $this->index($translationDocuments);
+            }
+        }
+    }
+
+    /**
+     * Index search engine documents corresponding to content object fields.
+     *
+     * Ported from the legacy code
+     * @see https://github.com/ezsystems/ezpublish-legacy/blob/master/kernel/search/plugins/ezsearchengine/ezsearchengine.php#L45
+     *
+     * @param \eZ\Publish\SPI\Search\Document $searchDocument
+     */
+    public function index(Document $searchDocument)
+    {
+        $documentMap = [];
+        foreach ($searchDocument->documents as $document) {
+            $documentMap[$document->languageCode][] = $document;
+        }
+
+        $indexArray = array();
+        $indexArrayOnlyWords = array();
+        $wordCount = 0;
+        $placement = 0;
+
+        foreach ($documentMap as $translationDocuments) {
+            foreach ($translationDocuments as $translation) {
+                /** @var \eZ\Publish\Core\Search\Legacy\Content\Document $translation */
+                $searchableFields = array_filter($translation->fields, function (Field $field) {
+                    return $field->type instanceof FullTextField;
+                });
+                foreach ($searchableFields as $field) {
+                    /** @var Field $field */
+                    $text = !is_array($field->value) ? $field->value : implode(' ', $field->value); // join Multiple* fields
+                    // Split text on whitespace
+                    if (is_numeric(trim($text))) {
+                        $integerValue = (int) $text;
+                        if ($integerValue > $this->dbMaxInt) {
+                            $integerValue = 0;
+                        }
+                    } else {
+                        $integerValue = 0;
+                    }
+                    $wordArray = explode(' ', $text);
+                    foreach ($wordArray as $word) {
+                        if (trim($word) != '') {
```

Comment:
rather return/continue early pattern, so:
and always strict checking when possible _(I'm assuming it is possible here in example below, but you'll know)_
```php
if (trim($word) === '') {
    continue;
}
```